src/odata-to-abstract-sql.ts

```diff
@@ -74,6 +74,12 @@ interface Resource extends AbstractSqlTable {
 	definition?: Definition;
 }
 
+export interface ResourceFunction {
+	(this: OData2AbstractSQL, property: any):
+		| BooleanTypeNodes
+		| { resource: Resource; name: string };
```

Comment:
Is this type defined anywhere else already?

Reply:
Not to my knowledge.

Reply:
The type definition came from here: https://github.com/balena-io-modules/odata-to-abstract-sql/pull/48/files#diff-3c71cc0b38f3742d3924bce55d96cf56R1034